qualtran/cirq_interop/_cirq_to_bloq.py

```diff
@@ -252,11 +275,27 @@ def _ensure_in_reg_exists(
     for qreg, soq in new_qreg_to_qvar.items():
         if len(in_reg_qubits) > 1 and qreg.qubits and qreg.qubits[0] in in_reg_qubits:
             assert len(qreg.qubits) == 1, "Individual qubits should have been split by now."
-            soqs_to_join[qreg.qubits[0]] = soq
+            # Need to split single-qubit Fxp registers which are ambiguous,
+            # these will be appropriately joined later.
+            if isinstance(qreg.dtype, QFxp):
+                soqs_to_join[qreg.qubits[0]] = bb.split(soq=soq)[0]
```

Comment:
Should we do a cast here instead? Why do we need to do this only for `QFxp` ? The input register of `Join` expects an input register of type `QBit()` so I imagine `soqs_to_join[qreg.qubits[0]] = soq` should correspond to a soquet of type `QBit` for every soquet that needs to be joined but we need to insert a special case only for `QFxp` ?

Reply:
Yeah you're right. Cast is much better.